lxd/storage_volumes.go

```diff
@@ -432,6 +686,29 @@ func storagePoolVolumesGet(d *Daemon, r *http.Request) response.Response {
 			}
 		}
 
+		if allPools {
+			poolNames, err := tx.GetStoragePoolNames(ctx)
+			if err != nil {
+				return fmt.Errorf("Failed to get storage volumes: %w", err)
+			}
+
+			for _, pool := range poolNames {
+				poolID, err := tx.GetStoragePoolID(ctx, pool)
+				if err != nil {
+					return fmt.Errorf("Failed to get storage volumes: %w", err)
+				}
+
+				poolVolumes, err := tx.GetStoragePoolVolumes(ctx, poolID, memberSpecific, filters...)
+				if err != nil {
+					return fmt.Errorf("Failed loading storage volumes: %w", err)
+				}
+
+				dbVolumes = append(dbVolumes, poolVolumes...)
+			}
+
+			return err
+		}
```

Comment:
I'm not opposed to this change, but perhaps it is worth writing a database method to get all storage volumes for all pools, rather than getting a list of storage pool names, then getting their IDs, then getting the storage volumes for each pool?

Reply:
Yeah this is on the cards already, at least for getting storage pool names and ids together.

Reply:
And we agreed to include that change in a separate PR.